webapp/plugins/index.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import store from 'stores/redux_store.jsx';
+
+global.window.plugins = {};
+global.window.plugins.components = {};
+
+export function registerComponents(components) {
+    global.window.plugins.components = {...components, ...global.window.plugins.components};
```

Comment:
Only thing I could see for having them in the redux store would be some sort of hot reloading behaviour. That shouldn't be too difficult though since we just have the `Pluggable` component listen to the redux store for changes